src/Adapter/Language/LanguageImageManager.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Language;
+
+use ImageType;
+
+/**
+ * Handles language images (flag, "no image" placeholders)
+ */
+class LanguageImageManager
```

Comment:
Isn't redundant to have `Language\Language`?

Reply:
I think that once imported it makes more sense to see a `LanguageImageManager` rather than an `ImageManager`. In this case the `Language` part is not a namespace or contextual information but part of the class role: "it handles language images", not "images in the context of a language".